File: app/src/main/java/com/zibete/proyecto1/utils/GlassEffect.java
package com.zibete.proyecto1.utils;

import android.graphics.drawable.AnimationDrawable;
import android.graphics.drawable.Drawable;
import android.view.View;

import eightbitlab.com.blurview.BlurTarget;
import eightbitlab.com.blurview.BlurView;

public class GlassEffect {


    public static void startGlowIfAny(View glowBorder) {
        if (glowBorder == null) return;
        Drawable bg = glowBorder.getBackground();
        if (bg instanceof AnimationDrawable) {
            AnimationDrawable anim = (AnimationDrawable) bg;
            anim.setEnterFadeDuration(600);
            anim.setExitFadeDuration(600);
            anim.start();
        }
    }

    /** Configura el efecto glass (BlurView 3.x con BlurTarget) */
    public static void applyGlassEffect(BlurView blurView, View itemView) {
        if (blurView == null) return;

        float radius = 16f;

        // El BlurTarget es el "contenedor base" a desenfocar
        BlurTarget blurTarget = new BlurTarget(itemView.getRootView().getContext());

        blurView.setupWith(blurTarget)
                .setBlurRadius(radius)
                .setBlurAutoUpdate(true);

        // sombreado suave para contraste
        blurView.setOverlayColor(0x1A000000);
    }


}

File: app/src/main/java/com/zibete/proyecto1/utils/GlassEffect.kt
package com.zibete.proyecto1.utils

import android.graphics.drawable.AnimationDrawable
import android.view.View
import android.view.ViewGroup
import eightbitlab.com.blurview.BlurView

object GlassEffect {

    fun startGlowIfAny(glowBorder: View?) {
        if (glowBorder == null) return
        val bg = glowBorder.background
        if (bg is AnimationDrawable) {
            bg.setEnterFadeDuration(600)
            bg.setExitFadeDuration(600)
            bg.start()
        }
    }

    /**
     * Configura el efecto glass usando BlurView 3.x (sin setBlurAlgorithm)
     */
    fun applyGlassEffect(blurView: BlurView?, itemView: View) {
        if (blurView == null) return

        val radius = 16f

        // Root sobre el que se va a tomar el contenido a desenfocar
        val rootView = itemView.rootView as? ViewGroup ?: return
        val windowBackground = rootView.background

//        blurView.setupWith(rootView)
//            .setFrameClearDrawable(windowBackground) // opcional
//            .setBlurRadius(radius)
//            .setBlurAutoUpdate(true)
//            .setHasFixedTransformationMatrix(true)

        // sombreado suave para contraste
        blurView.setOverlayColor(0x1A000000.toInt())
    }
}
